api/src/main/java/org/apache/iceberg/ManifestListFile.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.nio.ByteBuffer;
+
+public interface ManifestListFile {
+
+  /** Location of manifest list file. */
+  String location();
+
+  /**
+   * In encrypted tables, the manifest lists are encrypted. Returns key metadata of the encrypted
+   * manifest list file.
+   */
+  ByteBuffer keyMetadata();
+
+  /**
+   * The manifest list key metadata is encrypted with a "key encryption key" (KEK). Returns the KEK
+   * ID for this manifest file.
+   */
+  String metadataEncryptionKeyID();
+
+  /** Returns the manifest list key metadata, encrypted with its KEK. */
+  ByteBuffer encryptedKeyMetadata();
```

Comment:
I don't think that these should expose both encrypted and unencrypted versions of key metadata. If this object stores the encrypted key metadata, then it should only exposed the encrypted version. Similarly, other classes that store the unencrypted metadata should not be responsible for encrypting. These are simple classes and should stay that way rather than being responsible for encryption or carrying multiple versions of the same thing.